Fix for out of bounds registers_ access in mobile TorchScript interpreter

calvano-fb opened the pull request:
Summary:
The TorchScript interpreter had multiple opcodes whose logic had the potential to access the registers_ array out of bounds.

This change ensures that all registers_ accesses are in bounds or an exception will be thrown.

Test Plan: contbuild + OSS signals

Differential Revision: D49748737

malfet commented on torch/csrc/jit/mobile/interpreter.cpp:
```diff
@@ -386,6 +386,9 @@ bool InterpreterState::run(Stack& stack) {
 }
 
 IValue& InterpreterState::reg(size_t reg) {
+  if (registers_.empty() || reg == 0 || reg > registers_.size()) {
+    TORCH_CHECK(false, "Invalid register index: ", reg);
+  }
```
Nit (but also, please confirm the perf implication of this change, as this function can be on the hot path)
```suggestion
    TORCH_CHECK(!registers_.empty(), "Registers are empty");
    TORCH_CHECK(reg > 0 && reg =< registers_.size(), "Invalid register index: ", reg);
```

calvano-fb committed:
Fix for out of bounds registers_ access in mobile TorchScript interpreter (#110300)

Summary:

The TorchScript interpreter had multiple opcodes whose logic had the potential to access the registers_ array out of bounds.

This change ensures that all registers_ accesses are in bounds or an exception will be thrown.

Test Plan: contbuild + OSS signals

Reviewed By: malfet

Differential Revision: D49748737

malfet commented:
@pytorchbot merge

calvano-fb commented:
@pytorchbot merge
